pkg/controller/persistentvolume/types.go

```diff
@@ -94,6 +94,16 @@ func (pvIndex *persistentVolumeOrderedIndex) Find(searchPV *api.PersistentVolume
 	// potential matches (the GCEPD example above).
 	allPossibleModes := pvIndex.allPossibleMatchingAccessModes(searchPV.Spec.AccessModes)
 
+	// the searchPV should contain an annotation that allows pre-binding to a claim.
+	// we can use the same annotation value (pvc's namespace/name) and check against
+	// existing volumes to find an exact match. It is possible that a bind is made (ClaimRef persisted to PV)
+	// but the fail to update claim.Spec.VolumeName fails.  This check allows the claim to find the volume
+	// that's already bound to the claim.
+	preboundClaim := ""
```

Comment:
I believe I can remove the annotation-based pre-bind altogether, but that was merged earlier and this PR slightly changes the binding behavior.  I wasn't sure who might be affected by that, so I figured I would remove the annotation in the future when dynamic provisioning goes from experimental to real feature.